version: '3.8'

networks:
  taiza-resume-network:
    driver: bridge

volumes:
  db_data:

services:
  app:
    build:
      context: ./docker
      dockerfile: Dockerfile
    container_name: taiza-resume-app
    restart: unless-stopped
    working_dir: /var/www
    volumes:
      - ./:/var/www
    networks:
      - taiza-resume-network
    env_file:
      - ./.env

  postgres:
    image: postgres:13.4
    container_name: taiza-resume-postgres
    environment:
      POSTGRES_DB: ${DB_DATABASE}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
    volumes:
      - db_data:/var/lib/postgresql/data
      - ./docker/init-db.sql:/docker-entrypoint-initdb.d/init-db.sh
    networks:
      - taiza-resume-network
    ports:
      - "5432:5432"
    restart: unless-stopped

  nginx:
    image: nginx:stable-alpine
    container_name: taiza-resume-nginx
    ports:
      - "8082:80"
      - "444:443"
    volumes:
      - ./docker/nginx:/etc/nginx/conf.d
      - ./docker/nginx/default.conf:/etc/nginx/conf.d/default.conf
    depends_on:
      - app
    networks:
      - taiza-resume-network

